--- a/libweston/compositor-rdp.c
+++ b/libweston/compositor-rdp.c
@@ -618,13 +618,21 @@
 	context->rfx_context->mode = RLGR3;
 	context->rfx_context->width = client->settings->DesktopWidth;
 	context->rfx_context->height = client->settings->DesktopHeight;
+#if FREERDP_VERSION_MAJOR == 1 && FREERDP_VERSION_MINOR == 1
 	rfx_context_set_pixel_format(context->rfx_context, RDP_PIXEL_FORMAT_B8G8R8A8);
+#else
+	rfx_context_set_pixel_format(context->rfx_context, PIXEL_FORMAT_BGRA32);
+#endif
 
 	context->nsc_context = nsc_context_new();
 	if (!context->nsc_context)
 		goto out_error_nsc;
 
+#if FREERDP_VERSION_MAJOR == 1 && FREERDP_VERSION_MINOR == 1
 	nsc_context_set_pixel_format(context->nsc_context, RDP_PIXEL_FORMAT_B8G8R8A8);
+#else
+	nsc_context_set_pixel_format(context->nsc_context, PIXEL_FORMAT_BGRA32);
+#endif
 
 	context->encode_stream = Stream_New(NULL, 65536);
 	if (!context->encode_stream)
